src/vs/workbench/contrib/sandbox/electron-sandbox/sandbox.contribution.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from 'vs/base/common/lifecycle';
+import { IWorkbenchContributionsRegistry, IWorkbenchContribution, Extensions as WorkbenchExtensions } from 'vs/workbench/common/contributions';
+import { Registry } from 'vs/platform/registry/common/platform';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle';
+import { INativeHostService } from 'vs/platform/native/electron-sandbox/native';
+import { IProductService } from 'vs/platform/product/common/productService';
+import { IWindowsConfiguration, IWindowSettings } from 'vs/platform/window/common/window';
+
+interface IConfiguration extends IWindowsConfiguration {
+	window: IWindowSettings & { experimental?: { useSandbox?: boolean } };
+}
+
+export class SandboxExperimentHandler extends Disposable implements IWorkbenchContribution {
+
+	private useSandbox: boolean | undefined;
+
+	constructor(
+		@INativeHostService private readonly nativeHostService: INativeHostService,
+		@IProductService productService: IProductService,
+		@IConfigurationService configurationService: IConfigurationService
+	) {
+		super();
+
+		if (productService.quality !== 'stable') {
+			return; // experiment only applies to stable
+		}
+
+		this.onConfigurationChange(configurationService.getValue<IConfiguration>());
+		this._register(configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration('window.experimental.useSandbox')) {
+				this.onConfigurationChange(configurationService.getValue<IConfiguration>());
+			}
+		}));
+	}
+
+	private onConfigurationChange(configuration: IConfiguration): void {
+		const useSandbox = configuration.window.experimental?.useSandbox;
```

Comment:
I know I could check for source here and make sure its default, but I thought to make this super simple and just always react on the full setting value to simplify the code.

Reply:
Yeah though I wanted to keep it simple and just always reflect the settings state as it is in the renderer in the main process, irrespective how the value got there. I felt this approach is safer because it always reflects a consistent state and there is never a mismatch with either the default value or the configured value.